core/src/main/java/org/apache/calcite/sql/SqlDataTypeSpec.java

```diff
@@ -168,12 +163,8 @@ public Boolean getNullable() {
     return nullable;
   }
 
-  public Boolean getUppercase() {
-    return uppercase;
-  }
-
-  public Boolean getCaseSpecific() {
-    return caseSpecific;
+  public List<SqlColumnAttribute> getColumnAttributes() {
+    return columnAttributes;
```

Comment:
Both the uppercase and casespecific classes have their states stored in private fields with no getter method. So whereas before it was possible to directly access the uppercase and caseSpecific boolean fields via the getUppercase and getCaseSpecific methods, now the only way is to call getColumnAttributes(), unparse each SqlColumnAttribute, and then essentially parse it again. This might not be a problem if getUppercase and getCaseSpecific were never really used, but if there is a scenario in which the state of the column attributes needs to be accessed, it might be worth adding some getter methods to the ColumnAttributes.

Reply:
The getters before were only for unparsing purposes so I think it's fine to get rid of them for now. If they do need to get accessed at another point we can re-add them.